src/constructors/regular.c

```diff
@@ -311,12 +311,25 @@ int igraph_lattice(igraph_t *graph, const igraph_vector_t *dimvector,
 int igraph_ring(igraph_t *graph, igraph_integer_t n, igraph_bool_t directed,
                 igraph_bool_t mutual, igraph_bool_t circular) {
 
-    igraph_vector_t v = IGRAPH_VECTOR_NULL;
-
     if (n < 0) {
         IGRAPH_ERRORF("The number of vertices must be non-negative, got %" IGRAPH_PRId ".", n, IGRAPH_EINVAL);
     }
+    
+    if(n == 1 && circular)
+    {
+        //create the graph
+        IGRAPH_CHECK(igraph_small(graph,1,directed,0,0,-1));        
+        return IGRAPH_SUCCESS;
+    }
 
+    if(n == 2 && circular && directed)
+    {
+        //create the graph
+        IGRAPH_CHECK(igraph_small(graph,2,directed,0,1,1,0,-1));
+        return IGRAPH_SUCCESS;
+    }
+
+    igraph_vector_t v = IGRAPH_VECTOR_NULL;
```

Comment:
This declaration can stay at the top of the function. It also doesn't need to be initialized, so you can change it to `igraph_vector_t v;`

Reply:
That was already there when i cloned the repository. I can remove if you want.